typings/index.d.ts

```diff
@@ -0,0 +1,1909 @@
+  type Margin = 'none' | 'dense' | 'normal';
+}
+
+declare module 'material-ui' {
+  export { default as AppBar } from 'material-ui/AppBar';
+  export { default as Avatar } from 'material-ui/Avatar';
+  export { default as Badge } from 'material-ui/Badge';
+
+  export {
+    default as BottomNavigation,
+    BottomNavigationButton,
+  } from 'material-ui/BottomNavigation';
+
+  export { default as Button } from 'material-ui/Button';
+
+  export {
+    default as Card,
+    CardActions,
+    CardContent,
+    CardHeader,
+    CardMedia,
+  } from 'material-ui/Card';
+
+  export { default as Checkbox } from 'material-ui/Checkbox';
+  export { default as Chip } from 'material-ui/Chip';
+
+  export {
+    default as Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+  } from 'material-ui/Dialog';
+
+  export { default as Divider } from 'material-ui/Divider';
+  export { default as Drawer } from 'material-ui/Drawer';
+
+  export {
+    FormControl,
+    FormGroup,
+    FormLabel,
+    FormHelperText,
+    FormControlLabel,
+  } from 'material-ui/Form';
+
+  export { default as Hidden } from 'material-ui/Hidden';
+  export { default as Icon } from 'material-ui/Icon';
+  export { default as IconButton } from 'material-ui/IconButton';
+
+  export { default as Input, InputLabel } from 'material-ui/Input';
+
+  export { default as Grid } from 'material-ui/Grid';
+
+  export {
+    default as List,
+    ListItem,
+    ListItemAvatar,
+    ListItemIcon,
+    ListItemSecondaryAction,
+    ListItemText,
+    ListSubheader,
+  } from 'material-ui/List';
+
+  export { default as Menu, MenuItem, MenuList } from 'material-ui/Menu';
+
+  export { default as Paper } from 'material-ui/Paper';
+
+  export { CircularProgress, LinearProgress } from 'material-ui/Progress';
+
+  export { default as Radio, RadioGroup } from 'material-ui/Radio';
+
+  export { default as Snackbar, SnackbarContent } from 'material-ui/Snackbar';
+
+  export { MuiThemeProvider } from 'material-ui/styles';
+
+  import * as colors from 'material-ui/colors';
+  export { colors };
+
+  export { default as SvgIcon } from 'material-ui/SvgIcon';
+
+  export { default as Switch } from 'material-ui/Switch';
+
+  export {
+    default as Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    TableSortLabel,
+  } from 'material-ui/Table';
+
+  export { default as Tabs, Tab } from 'material-ui/Tabs';
+
+  export { default as Typography } from 'material-ui/Typography';
+
+  export { default as TextField } from 'material-ui/TextField';
+
+  export { default as Toolbar } from 'material-ui/Toolbar';
+}
+
+/* ============================================= */
+/*                                               */
+/*                   COMPONENTS                  */
+/*                                               */
+/* ============================================= */
+
+declare module 'material-ui/AppBar' {
```

Comment:
And re-create the `src` folder structure inside of the `typings` folder?
Guess it would also be possible to include `index.d.ts` files in the corresponding component folder. This would make the `typings` property in the `package.json` redudant.

I chose to declare modules inside a single file to "adhere" what was done with the `0.x` typings :)